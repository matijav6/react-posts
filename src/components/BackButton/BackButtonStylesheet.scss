$size: 65;

.back-button {
  height: $size + px;
  width: $size + px;
  position: fixed;
  border-radius: $size + px;
  right: 15px;
  bottom: 15px;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
}

.back-button > h5 {
  margin: 0;
  padding: 0;
}

.back-button:hover {
  background-color: #8dc5ff;
  transition: background-color .5s ease-out;
}
