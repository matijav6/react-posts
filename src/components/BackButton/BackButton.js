import React from 'react';
import {useHistory} from 'react-router-dom';
import './BackButtonStylesheet.scss';

export const BackButton = () => {
    const history = useHistory();

    return (
        <div onClick={() => history.goBack()} className={'back-button alert-primary'}>
            <h5>{'<-'}</h5>
        </div>
    );
};
