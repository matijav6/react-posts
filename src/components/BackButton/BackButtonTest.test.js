import renderer from 'react-test-renderer';
import {BackButton} from './BackButton'

test('Back button renders correctly', () => {
    const tree = renderer
        .create(<BackButton/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
