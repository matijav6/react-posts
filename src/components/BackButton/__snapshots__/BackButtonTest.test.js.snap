// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back button renders correctly 1`] = `
<div
  className="back-button alert-primary"
  onClick={[Function]}
>
  <h5>
    &lt;-
  </h5>
</div>
`;
